GameBoard/HostedServices/CronScheduledService.cs

```diff
@@ -37,12 +40,22 @@ private async Task RunIfDueAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task KeepAliveAsync(CancellationToken cancellationToken)
+        {
+            using (var scope = ServiceScopeFactory.CreateScope())
+            {
+                var keepAlive = scope.ServiceProvider.GetService<IKeepAlive>();
+                await keepAlive.KeepAliveAsync(cancellationToken);
+            }
```

Comment:
Why do we need to create this nested scope inside the scope already created in ScopedHostService?

Reply:
This scope isn't inside a scope created in `ScopedHostService`. That scope would be created in the `base.ExecuteAsync` call. I changed things around so that now everything that `CronScheduledService` does is in scope, including the keep-alive.